File: game-admin/src/main/java/com/dkm/config/GlobalCorsConfig.java
package com.dkm.config;

import org.jetbrains.annotations.NotNull;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.CorsConfigurationSource;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

/**
 * 处理全局
 */
@Configuration
public class GlobalCorsConfig {
    private CorsConfiguration addCorsConfig() {
        CorsConfiguration configuration = new CorsConfiguration();
        configuration.addAllowedOrigin("*");
        configuration.addAllowedHeader("*");
        configuration.addAllowedMethod("*");
        return configuration;
    }

    @Bean
    @NotNull
    public CorsFilter corsFilter() {
        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration("/**", this.addCorsConfig());
        return new CorsFilter(source);
    }
}

File: game-admin/src/main/java/com/dkm/config/GlobalCorsConfig.kt
package com.dkm.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter

/**
 * @author OpenE
 */
@Configuration
class GlobalCorsConfig {
    private fun addCorsConfig(): CorsConfiguration {
        return CorsConfiguration().run {
            addAllowedOrigin("*")
            addAllowedHeader("*")
            addAllowedMethod("*")
            this
        }
    }

    @Bean
    fun corsFilter(): CorsFilter {
        val source = UrlBasedCorsConfigurationSource()
        source.registerCorsConfiguration("/**", addCorsConfig())
        return CorsFilter(source)
    }
}